### PR TITLE
[ML] Include ML processor limits in `_ml/info` response

### PR DESCRIPTION
**Pull request by droberts195**
The `_ml/info` response now includes two extra fields in its `limits`:

1. `max_single_ml_node_processors`
2. `total_ml_processors`

These fields are _only_ included if they can be accurately calculated. If autoscaling is enabled and the ML nodes are not at their maximum size then these fields _cannot_ currently be accurately calculated. (This could potentially be improved in the future with additional settings set by the control plane.)

### CODE REVIEW

**Review by edsavage — APPROVED** [review]
LGTM

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java [hunk 1]
```diff
@@ -137,6 +145,38 @@ private Map<String, Object> limits() {
...
+                    * singleNodeProcessors.roundDown();
+                limits.put("total_ml_processors", totalMlProcessors.roundDown() + potentialExtraProcessors);
+            }
+        }
         return limits;
     }
+
+    static boolean areMlNodesBiggestSize(ByteSizeValue maxMLNodeSize, Collection<DiscoveryNode> mlNodes) {
+        if (maxMLNodeSize.getBytes() == 0) {
+            return true;
```
Comment (jonathan-buttner) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java:175]: When would `maxMLNodeSize` be zero? If the setting wasn't configured?

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java [hunk 2]
```diff
@@ -137,6 +145,38 @@ private Map<String, Object> limits() {
...
+            if (singleNodeProcessors.count() > 0) {
+                limits.put("max_single_ml_node_processors", singleNodeProcessors.roundDown());
+            }
+            Processors totalMlProcessors = MlProcessors.getTotalMlNodeProcessors(
+                nodes,
+                clusterSettings.get(MachineLearning.ALLOCATED_PROCESSORS_SCALE)
+            );
+            if (totalMlProcessors.count() > 0) {
+                int potentialExtraProcessors = Math.max(0, clusterSettings.get(MachineLearning.MAX_LAZY_ML_NODES) - mlNodes.size())
+                    * singleNodeProcessors.roundDown();
```
Comment (jonathan-buttner) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java:166]: It's ok for `singleNodeProcessors` to be zero here right?

**Review by jonathan-buttner — APPROVED** [review]

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java (hunk 1 above)
Comment (droberts195) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java:175]: Yes, zero is the default setting - set here: https://github.com/elastic/elasticsearch/blob/22381fd6a72687dbbaf3cfb76ce158625fc44b0c/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java#L698

In theory an operator could explicitly set zero, but we'd treat that the same as not set.

Comment (droberts195) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java:175]: We only expect this setting to be set in Cloud environments, and we assume ML nodes won't get scaled up automatically in self-managed environments.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java (hunk 2 above)
Comment (droberts195) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportMlInfoAction.java:166]: Yes, it's theoretically possible for a really small ML node running as a Docker container to have less than 1 full CPU to itself.

We won't allocate a model to that node unless it's started as "low priority". So it's OK to round down CPUs to zero for allocation purposes.

**Pull request merged**